R/namespace.R

```diff
@@ -16,5 +16,43 @@ namespace_imports <- function(path = find_package()) {
   data.frame(
     pkg = unlist(lapply(imports, function(x) rep(x[[1L]], length(x[[2L]])))),
     fun = unlist(lapply(imports, `[[`, 2L)),
-    stringsAsFactors = FALSE)
+    stringsAsFactors = FALSE
+  )
 }
+
+# filter namespace_imports() for S3 generics
+# this loads all imported namespaces
+imported_s3_generics <- function(ns_imports) {
+  is_generic <- vapply(
+    seq_len(nrow(ns_imports)),
+    function(i) {
+      fun_obj <- get(ns_imports$fun[i], envir = asNamespace(ns_imports$pkg[i]))
```

Comment:
we can do `mode='function'` in `get()`

Reply:
Are you sure? This would cause an error if any `ns_imports` doesn't point to a function.
I can imagine you could e.g. import a dataset. No?

Reply:
Ah, I'm wrong. I thought `get()` would give `NULL` or the like in that case, not error.